src/lib/Type.hs

```diff
@@ -636,8 +634,8 @@ flattenType :: Type -> [(BaseType, [Int])]
 flattenType ty = case ty of
   BaseType b  -> [(b, [])]
   RecType r -> concat $ map flattenType $ toList r
-  TabType (IdxSetLit n) a -> [(b, n:shape) | (b, shape) <- flattenType a]
-  IdxSetLit _ -> [(IntType, [])]
+  TabType (FixedIntRange low high) a -> [(b, (high - low):shape) | (b, shape) <- flattenType a]
+  FixedIntRange _ _ -> [(IntType, [])]
```

Comment:
I'm not sure if this is correct, so please check this carefully. IIUC `flattenType` is meant to give you something that can hold a runtime representation of a value of that type, and for indices (no matter what's the index set) it's just a single Int. If we were interested in serializing the type itself, then we would need at least two integers, but I don't think that this is what's happening here. Are the types saved in plain text?

Reply:
Yes, flattening is used in the binary serialization format, and the (unflattened) type itself is stored explicitly, using its concrete syntax. (e.g. you can see that we parse it [here](https://github.com/google-research/dex-lang/blob/master/src/lib/Serialize.hs#L109)). So it's fine that we lose the type information when we flatten.